packages/x-components/src/components/result/__tests__/base-result-add-to-cart.spec.ts

```diff
@@ -1,47 +1,86 @@
 
-  it('emits UserClickedResultAddToCart when the user click on the component', () => {
-    const resultAddToCartWrapper = mount(BaseResultAddToCart, {
-      localVue,
-      propsData: { result }
-    });
+  function renderAddToCart({
+    result = { name: 'TEST' } as Result,
+    template = '<BaseResultAddToCart :result="result"/>',
+    methods = {}
+  }: RenderAddToCartOptions): RenderAddToCartApi {
+    const [, localVue] = installNewXPlugin();
+    const wrapper = mount(
+      { template, data: () => ({ result }), methods },
+      {
+        components: { BaseResultAddToCart },
+        localVue
+      }
+    );
+    return {
+      clickAddToCart(): Promise<void> {
+        return wrapper
+          .find(getDataTestSelector('result-add-to-cart'))
+          .trigger('click') as Promise<void>;
+      },
+      on: wrapper.vm.$x.on,
+      addToCartWrapper: wrapper.find(getDataTestSelector('result-add-to-cart'))
+    };
+  }
+
+  it('emits UserClickedResultAddToCart when the user click on the component', async () => {
+    const { clickAddToCart, on } = renderAddToCart({ result });
     const listener = jest.fn();
-    resultAddToCartWrapper.vm.$x.on('UserClickedResultAddToCart').subscribe(listener);
-    resultAddToCartWrapper.trigger('click');
+    on('UserClickedResultAddToCart').subscribe(listener);
+    await clickAddToCart();
     expect(listener).toHaveBeenCalledWith(result);
   });
 
   it('renders the content overriding default slot', () => {
-    const wrapperComponent = {
+    const { addToCartWrapper } = renderAddToCart({
       template: `
         <BaseResultAddToCart :result="result">
           <img data-test="result-add-to-cart-icon" src="./add-to-cart.svg" />
           <span data-test="result-add-to-cart-text">Add to cart</span>
         </BaseResultAddToCart>
       `,
-      props: ['result'],
-      components: {
-        BaseResultAddToCart
-      }
-    };
-
-    const customResultAddToCartWrapper = mount(wrapperComponent, {
-      localVue,
-      propsData: { result }
+      result
     });
+
+    expect(addToCartWrapper.element).toBeDefined();
     expect(
-      customResultAddToCartWrapper.find(getDataTestSelector('result-add-to-cart')).element
-    ).toBeDefined();
-    expect(
-      customResultAddToCartWrapper.find(getDataTestSelector('result-add-to-cart-icon')).element
+      addToCartWrapper.find(getDataTestSelector('result-add-to-cart-icon')).element
     ).toBeDefined();
-    expect(customResultAddToCartWrapper.text()).toEqual('Add to cart');
+    expect(addToCartWrapper.text()).toEqual('Add to cart');
+  });
+
+  it('uses the listeners passed', async () => {
+    const listener = jest.fn();
+    const { clickAddToCart } = renderAddToCart({
+      template: '<BaseResultAddToCart @click="miClick" :result="result"/>',
+      result,
+      methods: {
+        miClick: listener
+      }
+    });
+    await clickAddToCart();
+
+    expect(listener).toHaveBeenCalled();
```

Comment:
What is the expected payload for this click callback? A result? A `MouseEvent` (which it is right now)?

```suggestion
    expect(listener).toHaveBeenCalledTimes(1);
    expect(listener).toHaveBeenCalledWith(expect.any(MouseEvent));
```

Reply:
👍🏻